### PR TITLE
Initial pass at Firestore Watch documentation

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Snippets/UserGuideSnippets.cs [hunk 1]
```diff
@@ -355,12 +355,19 @@ public class City
...
             FirestoreChangeListener listener = query.Listen(snapshot =>
             {
-                Console.WriteLine($"Callback received snapshot");
+                Console.WriteLine($"Callback received query snapshot");
                 Console.WriteLine($"Count: {snapshot.Count}");
                 Console.WriteLine("Changes:");
                 foreach (DocumentChange change in snapshot.Changes)
                 {
-                    Console.WriteLine($"{change.Document.Reference.Id}: ChangeType={change.ChangeType}; OldIndex={change.OldIndex}; NewIndex={change.NewIndex})");
+                    var document = change.Document;
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Snippets/UserGuideSnippets.cs]: Please don't use `var`.

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Snippets/UserGuideSnippets.cs [hunk 2]
```diff
@@ -355,12 +355,19 @@ public class City
...
+                    Console.WriteLine($"{document.Reference.Id}: ChangeType={change.ChangeType}; OldIndex={change.OldIndex}; NewIndex={change.NewIndex})");
+                    if (document.Exists)
+                    {
+                        var name = document.GetValue<string>("Name");
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Snippets/UserGuideSnippets.cs]: And here.
I know it's very obvious what the types are here, but still think we should not use `var` anywhere in documentation.

**Comment by jskeet** [comment]
(Have notified Firestore folks about this; we may wait for their input before merging.)

**Comment by jskeet** [comment]
Merging to avoid this getting stale. Can still revisit later.

**Pull request merged**